library/src/main/java/com/novoda/downloadmanager/MigrationExtractor.java

```diff
@@ -68,8 +70,13 @@
                     while (downloadsCursor.moveToNext()) {
                         String originalFileId = downloadsCursor.getString(FILE_ID_COLUMN);
                         String originalNetworkAddress = downloadsCursor.getString(NETWORK_ADDRESS_COLUMN);
-                        String originalFileLocation = downloadsCursor.getString(FILE_LOCATION_COLUMN);
-                        String sanitizedOriginalFileLocation = MigrationStoragePathSanitizer.sanitize(originalFileLocation);
+
+                        // hint: file:/data/user/0/com.channel4.ondemand/files/all4/23241337
+                        String originalUniqueFileLocation = downloadsCursor.getString(FILE_UNIQUE_LOCATION_COLUMN);
+                        String sanitizedOriginalUniqueFileLocation = MigrationStoragePathSanitizer.sanitize(originalUniqueFileLocation);
+
+                        // _data: /data/user/0/com.channel4.ondemand/files/all4/23241337-1
+                        String originalFileLocation = downloadsCursor.getString(FILE_ORIGINAL_LOCATION_COLUMN);
```

Comment:
Did you double check that this one is always populated? I seem to remember that we had issues because sometimes it was empty (and that's the reason why we moved to `hint`)

Reply:
This is fine for **Completed Downloads** the `_data` is only populated when a download is complete 😄 I did hesitate changing this originally but decided to so that it matched the `PartialMigrationExtractor`, now I see I shouldn't have bothered 😬